src/core/account/plugin-api.js

```diff
@@ -59,6 +61,41 @@ export class CurrencyConfig extends Bridgeable<EdgeCurrencyConfig> {
     return state.accounts[accountId].customTokens[pluginId]
   }
 
+  async addCustomToken(token: EdgeToken): Promise<string> {
+    const { _accountId: accountId, _ai: ai, _pluginId: pluginId } = this
+    const tokenId = await getTokenId(ai, pluginId, token)
+
+    ai.props.dispatch({
+      type: 'ACCOUNT_CUSTOM_TOKEN_ADDED',
+      payload: { accountId, pluginId, tokenId, token }
+    })
+    return tokenId
+  }
+
+  async changeCustomToken(tokenId: string, token: EdgeToken): Promise<void> {
+    const { _accountId: accountId, _ai: ai, _pluginId: pluginId } = this
+    const newTokenId = await getTokenId(ai, pluginId, token)
+    if (newTokenId !== tokenId) {
+      throw new Error(
+        `The tokenId would change from ${tokenId} to ${newTokenId}`
```

Comment:
Maybe:

```js
throw new Error(`Mismatching tokenId '${newTokenId}' when updating for tokenId '${tokenId}'`)
```

or similar

Reply:
I think the existing message is better, because something has changed (like the contract address) that affects the token's identity. Otherwise, we could say that this is actually OK, drop the error message, and just treat this as an add / remove operation.